crates/cheatnet/tests/cheatcodes/declare.rs

```diff
@@ -1,95 +1,86 @@
     let (mut blockifier_state, _) = create_cheatnet_state(&mut cached_state);
 
-    let contract = "HelloStarknet".to_owned().to_felt252();
+    let contract = contract_name.to_owned().to_felt252();
     let contracts = get_contracts();
 
     let class_hash = blockifier_state.declare(&contract, &contracts).unwrap();
+    let expected_class_hash = class_hash_from_str(contract_name, &contracts);
 
-    assert_eq!(
-        class_hash,
-        ClassHash(stark_felt!(HELLO_STARKNET_CLASS_HASH))
-    );
+    assert_eq!(class_hash, expected_class_hash);
 }
 
-#[test]
-fn declare_multiple() {
+#[test_case(vec!["HelloStarknet", "ConstructorSimple"])]
+fn declare_multiple(contract_names: Vec<&str>) {
     let mut cached_state = create_cached_state();
     let (mut blockifier_state, _) = create_cheatnet_state(&mut cached_state);
 
-    let contract = "HelloStarknet".to_owned().to_felt252();
     let contracts = get_contracts();
 
-    let class_hash = blockifier_state.declare(&contract, &contracts).unwrap();
-
-    let contract = "ConstructorSimple".to_owned().to_felt252();
-
-    let class_hash2 = blockifier_state.declare(&contract, &contracts).unwrap();
-
-    assert_eq!(
-        class_hash,
-        ClassHash(stark_felt!(HELLO_STARKNET_CLASS_HASH))
-    );
-    assert_eq!(
-        class_hash2,
-        ClassHash(stark_felt!(CONSTRUCTOR_SIMPLE_CLASS_HASH))
-    );
+    for contract_name in contract_names {
+        let contract = contract_name.to_owned().to_felt252();
+        let class_hash = blockifier_state.declare(&contract, &contracts).unwrap();
+        let expected_class_hash = class_hash_from_str(contract_name, &contracts);
+        assert_eq!(class_hash, expected_class_hash);
+    }
 }
 
-#[test]
-fn declare_same_contract() {
+#[test_case("HelloStarknet", "is already declared")]
+fn declare_same_contract(contract_name: &str, error_msg: &str) {
     let mut cached_state = create_cached_state();
     let (mut blockifier_state, _) = create_cheatnet_state(&mut cached_state);
 
-    let contract = "HelloStarknet".to_owned().to_felt252();
+    let contract = contract_name.to_owned().to_felt252();
     let contracts = get_contracts();
 
     let class_hash = blockifier_state.declare(&contract, &contracts).unwrap();
-    assert_eq!(
-        class_hash,
-        ClassHash(stark_felt!(HELLO_STARKNET_CLASS_HASH))
-    );
-
-    let contract = "HelloStarknet".to_owned().to_felt252();
+    let expected_class_hash = class_hash_from_str(contract_name, &contracts);
+    assert_eq!(class_hash, expected_class_hash);
 
     let output = blockifier_state.declare(&contract, &contracts);
 
     assert!(match output {
         Err(CheatcodeError::Unrecoverable(EnhancedHintError::Anyhow(msg))) => {
-            msg.to_string().contains("is already declared")
+            msg.to_string().contains(error_msg)
         }
         _ => false,
     });
 }
 
-#[test]
-fn declare_non_existant() {
+#[test_case("GoodbyeStarknet")]
```

Comment:
Why bother with that if it's not multiple contracts parametrized here?

Reply:
It's clearer to see the main constant arg at the top imo, should it be reverted?